src/vanilla/Model/MethodGroupTS.cs

```diff
@@ -26,6 +26,74 @@ protected MethodGroupTS(string name): base(name)
         [JsonIgnore]
         public IEnumerable<MethodTS> MethodTemplateModels => Methods.Cast<MethodTS>();
 
+        public ISet<string> OperationModelNames
+        {
+            get
+            {
+                ISet<string> modelNames = new HashSet<string>();
+                while (true)
```

Comment:
What's the purpose of looping here? Generally if you have a stack of ModelTypes that you're iterating through (and adding to while you iterate), then you need a loop to check if the stack is empty. It looks like you're doing that graph traversal recursively, however, which indicates to me that you don't need a loop here.

Reply:
We want to ensure that if any polymorphic subtypes were discovered, that we 1. add any unknown property types to the closure, and 2. add any polymorphic subtypes of those newly discovered types to the closure.

Expressing this recursively (I think) would involve factoring out another method which consumes the initial size of the set, then performs another recursive call if the size of the set changed.

```cs
void FindModelNames(ISet<string> names, int initialSize, CompositeType model)
{
  ModelsClosure(names, model);
  if (initialSize != names.Count) FindModelNames(names, names.Count, model);
}
```

I think this is a little less clear than the `while (true) { ... }`. I might be missing something about this algorithm or how to simplify it, though! I think I would also be alright with making this explicitly stack-based instead for some kind of stylistic consistency.